web-local/src/lib/util/get-table-references/index.ts

```diff
@@ -0,0 +1,25 @@
+export interface Reference {
+  reference: string;
+  type: "from" | "join";
+  index: number;
+  referenceIndex: number;
+}
+
+/**
+ *
+ * @param sql the sql query
+ * @returns {Reference[]} an array of reference locations
+ */
+export function getTableReferences(sql: string) {
+  if (!sql) return [];
+  // eslint-disable-next-line no-useless-escape
+  const regex = /(?:from|join)\s+([a-zA-z0-9_.]+|"[a-zA-z0-9\.\-_\/:\s]+")/gim;
```

Comment:
Would this be able to extract the tables from a CTE? Would be good to have those tests from the previous suite.

Reply:
yes, it should work in the case of CTEs as well. There's actually nothing special about CTEs with this approach. I can add those tests though for the sake of clarity

Reply:
I've added a CTE test to the suite.